### DISCUSSION TITLE
Is there a way to insert non editable whitespace into textbox object ?

### DISCUSSION THREAD

**Discussion by Balbalnom** [fabricjs/fabric.js#8369] [General]
I am trying to do a text wrap feature similar to InDesign. 
ref: https://helpx.adobe.com/indesign/using/text-wrap.html#:~:text=Options%20dialog%20box.-,To%20display%20the%20Text%20Wrap%20panel%2C%20choose%20Window%20%3E%20Text%20Wrap,the%20wrap%20within%20the%20frame.

My general approach is coming up with my collision logic and inserting some white space during the `_splitTextIntoLines` process. And make this process compatible with dynamic width per line since fabric textbox also uses unified textbox width.
The main problem with this approach is when coming to the text box edit mode, I do not find an excellent solution to deal with the empty white spaces. Is there a possible way to insert noneditable white space into the textbox during edit mode?

Thanks!

**Comment by ShaMan123** [comment]
I wouldn't solve this with ws.
Did you look into the `path` option on text?
That's the way forward IMO

**Reply by Balbalnom** [reply]
But I do want to make the text behavior like a textbox, at least editable with customized styling and font.
